File: Insertion_Sort.py
# insertion sort

def insertion_sort(L):
    n = len(L)
    for i in range(1, n):
        pos = i
        while pos > 0 and L[pos] < L[pos - 1]:
            temp = L[pos]
            L[pos] = L[pos - 1]
            L[pos - 1] = temp
            pos = pos - 1
    return L

File: Insertion_Sort.c
// insertion sort: best case: O(n)
//                 worst case: O(n^2)

void insertion_sort(int arr[], int len) {
    for (int i = 1; i < len; i++) {
        for (int j = i; j > 0; j--) {
            if (arr[j] < arr[j - 1]) {
                // swap
                int temp = arr[j];
                arr[j] = arr[j - 1];
                arr[j - 1] = temp;
            }
        }
    }
}
